File: ros_tutorial_py/scripts/talker.py
#!/usr/bin/env python
'''
  Filename: talker.py
  Author: Ryan Shim
'''

import rospy
from std_msgs.msg import Int8

def talker():
    rospy.init_node('talker', anonymous=True)
    count_pub = rospy.Publisher('count', Int8, queue_size=10)
    rate = rospy.Rate(10) # 10Hz
    count = Int8()
    count.data = 0
    while not rospy.is_shutdown():
        rospy.loginfo("count: %d", count.data)
        count_pub.publish(count)
        rate.sleep()
        count.data += 1

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: ros_tutorial_cpp/src/talker.cpp
/*
  Filename: talker.cpp
  Author: Ryan Shim
*/

#include <ros/ros.h>
#include <std_msgs/Int8.h>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "talker");
  ros::NodeHandle nh;
  ros::Publisher count_pub = nh.advertise<std_msgs::Int8>("count", 10);
  ros::Rate rate(10); // 10Hz
  std_msgs::Int8 count;
  count.data = 0;
  while (ros::ok())
  {
    ROS_INFO("count: %d", count.data);
    count_pub.publish(count);
    ros::spinOnce();
    rate.sleep();
    count.data++;
  }
  return 0;
}
